src/main/java/com/jfrog/xray/client/impl/XrayClient.java

```diff
@@ -84,6 +91,10 @@ public CloseableHttpResponse head(String uri) throws IOException {
     }
 
     public CloseableHttpResponse post(String uri, Object payload) throws IOException {
+        return post(uri, payload, this.mapper);
+    }
+
+    public CloseableHttpResponse post(String uri, Object payload, ObjectMapper mapper) throws IOException {
```

Comment:
It looks like this method can be made private, and the signature above that does not get the mapper, can be used by external classes.